ts/src/test/Exchange/test.createOrder.ts

```diff
@@ -0,0 +1,228 @@
+            const minimumCostForBuy = getMinimumCostForBuy (exchange, market);
+            priceForMarketSellOrder = (minimumCostForBuy / amountToSell) * limitPriceSafetyMultiplierFromMedian;
+        }
+        const sellOrder = await testCreateOrder_submitSafeOrder (exchange, symbol, 'market', 'sell', amountToSell, priceForMarketSellOrder, params, skippedProperties);
+        const sellOrder_fetched = await testSharedMethods.tryFetchOrder (exchange, symbol, sellOrder['id'], skippedProperties);
+        // try to test that order was fully filled
+        const isClosedFetched = testSharedMethods.confirmOrderState (exchange, sellOrder_fetched, 'closed');
+        const isOpenFetched = testSharedMethods.confirmOrderState (exchange, sellOrder_fetched, 'open');
+        assert (isClosedFetched || isOpenFetched === undefined, warningPrefix + ' ' +  exchange.id + ' ' + symbol + ' order should be filled, but it is not. ' + exchange.json (sellOrder_fetched));
+        verboseOutput (exchange, symbol, 'SCENARIO 2 PASSED !!!');
+    } catch (e) {
+        throw new Error ('failed for Scenario 2: ' + e.toString ());
+    }
+    // *********** [Scenario 2 - END ] *********** //
+
+
+    // ***********
+    // above, we already tested 'limit' and 'market' orders. next, 'todo' is to create tests for other unified scenarios (spot, swap, trigger, positions, stoploss, takeprofit, etc)
+    // ***********
+}
+
+// ----------------------------------------------------------------------------
+
+async function testCreateOrder_cancelOrder (exchange, symbol, orderId = undefined) {
+    // cancel the order (one of the below methods is guaranteed to be existent, as this was checked in the start of this test)
+    let usedMethod = '';
+    let cancelResult = undefined;
+    if (exchange.has['cancelOrder'] && orderId !== undefined) {
+        usedMethod = 'cancelOrder';
+        cancelResult = await exchange.cancelOrder (orderId, symbol);
+    } else if (exchange.has['cancelAllOrders']) {
+        usedMethod = 'cancelAllOrders';
+        cancelResult = await exchange.cancelAllOrders (symbol);
+    } else if (exchange.has['cancelOrders']) {
+        usedMethod = 'cancelOrders';
+        if (orderId === undefined) {
+            cancelResult = await exchange.cancelOrders ([], symbol);
+        } else {
+            cancelResult = await exchange.cancelOrders ([ orderId ], symbol);
+        }
+    }
+    verboseOutput (exchange, symbol, 'canceled order using', usedMethod, ' : ', exchange.json (cancelResult));
+}
+
+// ----------------------------------------------------------------------------
+
+
+// ----------------------------------------------------------------------------
+
+async function testCreateOrder_submitSafeOrder (exchange, symbol, orderType, side, amount, price = undefined, params = {}, skippedProperties = {}) {
```

Comment:
testCreateOrder_submitSafeOrder this name makes no sense and mixes camelCase with snakeCase

Reply:
I've done that intentionally, specifically for PHP, because these method name pollute global namespace, so we should have somewhat "prefix" to avoid possible collision from other files, so, I agree, not a good format, but if so, we need to have `testCreateOrderSubmitSafeOrder` then

Reply:
it doesn't matter what is name (be it `submitSafeOrder` as example) but if that method name appears anywhere across whole project, the function name collision happens, so lmk what prefix/format should we use.

Reply:
that underscore was "conceptual divisor" in my thought, but might not be good, i agree

Reply:
ok, removed underscore